src/KPM.jl

```diff
@@ -67,6 +88,28 @@ function _addmomenta!(μlist, ket, h, bandbracket)
     return μlist
 end
 
+function _addmomenta!(μlist, ket, h, bandbracket, obs::AbstractMatrix)
+    order = length(μlist) - 1
+    ket0 = ket
+    ketL = similar(ket)
+    ket1 = similar(ket)
+    ket2 = similar(ket)
+    mul!(ketL', ket', obs) 
+    mulscaled!(ket1, h, ket0, bandbracket)
+    μlist[1] = real(ketL' * ket0)
+    μlist[2] += real(ketL' * ket1)
+    @showprogress for n in 3:1:(order+1) 
+        mulscaled!(ket2, h, ket1, bandbracket)
+        @. ket2 = 2 * ket2 - ket0
+        μlist[n] += real(ketL' * ket2)
```

Comment:
We can generalize this to any kind of `h` (even non-Hermitian) and for matrix kets just by changing all the `real(ket1' * ket2)` with a `proj(ket1, ket2) = tr(ket1' * ket2)`, with possible added methods as optimizations.